interfaces/builtin/storage_framework_service_test.go

```diff
@@ -0,0 +1,108 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/builtin"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/snap/snaptest"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type StorageFrameworkServiceInterfaceSuite struct {
+	iface       interfaces.Interface
+	coreSlot    *interfaces.Slot
+	classicSlot *interfaces.Slot
+	plug        *interfaces.Plug
+}
+
+var _ = Suite(&StorageFrameworkServiceInterfaceSuite{})
+
+func (s *StorageFrameworkServiceInterfaceSuite) SetUpTest(c *C) {
+	// a storage-framework-service slot on a storage-framework-service snap
+	const storageFrameworkServiceMockCoreSlotSnapInfoYaml = `name: storage-framework-service
+version: 1.0
+apps:
+ app:
+  command: foo
+  slots: [storage-framework-service]
+`
+	// a storage-framework-service slot on the core snap (as automatically added on classic)
+	const storageFrameworkServiceMockClassicSlotSnapInfoYaml = `name: core
+type: os
+slots:
+ storage-framework-service:
+  interface: storage-framework-service
+`
```

Comment:
You should drop storageFrameworkServiceMockClassicSlotSnapInfoYaml because the storage-framework-service interface is not an implicit interface and the basedeclaration says only snaps of type 'app' may slot this interface.